test/integration/timeout/timeout_stack_test.go

```diff
@@ -0,0 +1,60 @@
+package timeout_test
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework/manifest"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/integration/timeout"
+	. "github.com/onsi/ginkgo"
+	"time"
+)
+
+var _ = Describe("timeout feature test", func() {
+	var (
+		ctx context.Context
+		f   *framework.Framework
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		f = framework.New(framework.GlobalOptions)
+	})
+
+	Context("timeout default test dimensions", func() {
+		var stackPrototype timeout.TimeoutStack
+		var stacksPendingCleanUp []*timeout.TimeoutStack
+
+		BeforeEach(func() {
+			stacksPendingCleanUp = nil
+		})
+
+		AfterEach(func() {
+			for _, stack := range stacksPendingCleanUp {
+				stack.CleanupTimeoutStack(ctx, f)
+			}
+		})
+
+		for _, sdType := range []manifest.ServiceDiscoveryType{manifest.DNSServiceDiscovery} {
+			func(sdType manifest.ServiceDiscoveryType) {
+				It(fmt.Sprintf("Should behave correctly with and without timeout configured"), func() {
+					stackPrototype.ServiceDiscoveryType = sdType
+					stackDefault := stackPrototype
+
+					By("deploy timeout stack into cluster", func() {
+						stacksPendingCleanUp = append(stacksPendingCleanUp, &stackDefault)
+						stackDefault.DeployTimeoutStack(ctx, f)
+					})
+
+					By("sleep 30 seconds for Envoys to be configured", func() {
```

Comment:
Can we please add a TODO here to remove this once readiness probe PR https://github.com/aws/aws-app-mesh-controller-for-k8s/pull/325 is merged

Reply:
Readiness probe might not entirely take care of it here right?

Reply:
It can help make it more deterministic. Once the readiness probe is there and if Envoy is not ready, the container and pod condition `Ready` and `ContainersReady` will be set to false. You could query that (with a max # of retries) and move forward when pod is ready

Here's a sample of status->conditions for a pod which has readiness probe failing:
```
        "conditions": [
            {
                "lastProbeTime": null,
                "lastTransitionTime": "2020-07-31T22:10:55Z",
                "status": "True",
                "type": "Initialized"
            },
            {
                "lastProbeTime": null,
                "lastTransitionTime": "2020-07-31T22:10:55Z",
                "message": "containers with unready status: [envoy]",
                "reason": "ContainersNotReady",
                "status": "False",
                "type": "Ready"
            },
            {
                "lastProbeTime": null,
                "lastTransitionTime": "2020-07-31T22:10:55Z",
                "message": "containers with unready status: [envoy]",
                "reason": "ContainersNotReady",
                "status": "False",
                "type": "ContainersReady"
            },
            {
                "lastProbeTime": null,
                "lastTransitionTime": "2020-07-31T22:10:55Z",
                "status": "True",
                "type": "PodScheduled"
            }
        ],
```

Reply:
No, I mean we are not waiting here for Envoy to be ready and responsive here but more for the config to be populated from appmesh control plane. We can discuss..